### PR TITLE
Introduces XSS sink detecting eslint rules

### PR DESCRIPTION
**Pull request by LukeWood**
These eslint rules attempt to surface XSS sinks from eslint.
This commit is a first pass on
https://github.com/eclipse-theia/theia/issues/8398

Change-Id: I142bbba9785c4567dfbea1380f5a980560cf7413

#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: configs/xss.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "extends": ["plugin:no-unsanitized/DOM"],
+  "plugins": ["no-unsanitized"],
```
Comment (marcdumais-work) [inline configs/xss.eslintrc.json:3]: I get a warning about duplicate key here. Was having two `plugins` entries wanted here?

**Comment by marcdumais-work** [comment]
I am not an expert, but I was able to confirm that, if I re-introduce a XSS problem we previously fixed, it's picked-up by the new linting rule: 

Peek 2020-09-09 12-30

Other than the one pending comment, LGTM. 

@marechal-p Do you want to have a look?

**Comment by marcdumais-work** [comment]
BTW, the two dev-dependencies added though this PR are good, license-wise, according to the Foundation's approved license list.

File: configs/xss.eslintrc.json (hunk 1 above)
Comment (LukeWood) [reply configs/xss.eslintrc.json]: oops - resolved

Comment (LukeWood) [reply configs/xss.eslintrc.json]: fixed + resolved

**Comment by LukeWood** [comment]
Should be good to go!  fixed the 1 comment.

**Review by marechal-p — APPROVED** [review]
I was wondering if an error was better, but we can always upgrade later. LGTM.

**Comment by LukeWood** [comment]
> I was wondering if an error was better, but we can always upgrade later. LGTM.

IIRC we'd have a few errors in the build if it was error rather than warning ATM.

**Comment by marcdumais-work** [comment]
Thanks @LukeWood for this useful contribution! Merging.

**Pull request merged**